code/modules/organs/external/_external.dm

```diff
@@ -1003,90 +1024,83 @@ Note that amputating the affected organ does in fact remove the infection from t
-			victim.refresh_visible_overlays()
-		if(!QDELETED(src))
+			// Throw limb around.
+			if(src && isturf(loc))
+				throw_at(get_edge_target_turf(src, pick(global.alldirs)), rand(1,3), THROWFORCE_GIBS)
 			set_dir(SOUTH, TRUE)
-
-	switch(disintegrate)
-		if(DISMEMBER_METHOD_EDGE)
-			compile_icon()
-			add_blood(victim)
-			set_rotation(rand(180))
-			forceMove(get_turf(src))
-			if(!clean)
-				// Throw limb around.
-				if(src && isturf(loc))
-					throw_at(get_edge_target_turf(src, pick(global.alldirs)), rand(1,3), THROWFORCE_GIBS)
-				set_dir(SOUTH, TRUE)
-		if(DISMEMBER_METHOD_BURN, DISMEMBER_METHOD_ACID)
-			if(disintegrate == DISMEMBER_METHOD_BURN)
-				new /obj/effect/decal/cleanable/ash(get_turf(victim))
-			else
-				new /obj/effect/decal/cleanable/mucus(get_turf(victim))
-			for(var/obj/item/I in src)
-				if(I.w_class > ITEM_SIZE_SMALL && !istype(I,/obj/item/organ))
+	else
+		// Other attacks can destroy the limb entirely and place an item or decal.
+		var/atom/movable/gore = place_remains_from_dismember_method(disintegrate)
+		if(gore)
+			if(disintegrate == DISMEMBER_METHOD_BURN || disintegrate == DISMEMBER_METHOD_ACID)
+				for(var/obj/item/I in src)
+					if(I.w_class > ITEM_SIZE_SMALL && !istype(I,/obj/item/organ))
+						I.dropInto(loc)
+			else if(disintegrate == DISMEMBER_METHOD_BLUNT)
+				gore.throw_at(get_edge_target_turf(src,pick(global.alldirs)), rand(1,3), THROWFORCE_GIBS)
+				for(var/obj/item/organ/I in internal_organs)
+					I.do_uninstall() //No owner so run uninstall directly
+					I.dropInto(get_turf(loc))
+					if(!QDELETED(I) && isturf(loc))
+						I.throw_at(get_edge_target_turf(src,pick(global.alldirs)), rand(1,3), THROWFORCE_GIBS)
+				for(var/obj/item/I in src)
 					I.dropInto(loc)
-			qdel(src)
-		if(DISMEMBER_METHOD_BLUNT)
-			var/obj/gore
-			if(BP_IS_CRYSTAL(src))
-				gore = new /obj/item/shard(get_turf(victim), /decl/material/solid/gemstone/crystal)
-			else if(BP_IS_PROSTHETIC(src))
-				gore = new /obj/effect/decal/cleanable/blood/gibs/robot(get_turf(victim))
-			else
-				gore = new /obj/effect/decal/cleanable/blood/gibs(get_turf(victim))
-				if(species)
-					var/obj/effect/decal/cleanable/blood/gibs/G = gore
-					G.fleshcolor = use_flesh_colour
-					G.basecolor =  use_blood_color
-					G.update_icon()
-
-			gore.throw_at(get_edge_target_turf(src,pick(global.alldirs)), rand(1,3), THROWFORCE_GIBS)
-
-			for(var/obj/item/organ/I in internal_organs)
-				I.do_uninstall() //No owner so run uninstall directly
-				I.dropInto(get_turf(loc))
-				if(!QDELETED(I) && isturf(loc))
 					I.throw_at(get_edge_target_turf(src,pick(global.alldirs)), rand(1,3), THROWFORCE_GIBS)
+			if(!QDELETED(src))
+				qdel(src)
 
-			for(var/obj/item/I in src)
-				I.dropInto(loc)
-				I.throw_at(get_edge_target_turf(src,pick(global.alldirs)), rand(1,3), THROWFORCE_GIBS)
-
-			qdel(src)
+/obj/item/organ/external/proc/handle_async_post_dismemberment(var/mob/living/carbon/human/victim)
+	set waitfor = FALSE
+	sleep(1)
```

Comment:
It was awful before...and it's still awful. Just do it synchronously at the end. If you have performance concerns, add an arg to the proc that suppresses this stuff and then call it at the end of batch removal outside of qdel instead (similarly for recursive calls).

Notably, this spawn approach does _not_ address any performance concerns _because this is queued for every organ removed anyway_.

Reply:
I don't have performance concerns I just had no idea what this would impact if I didn't replicate the previous behavior.

Reply:
I actually never notice that spawn before... There's gotta be a way to get rid of it?

Reply:
It wasn't actually needed.